DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -576,7 +577,33 @@ class TrackerTopology {
   std::string print(DetId detid) const;
 
   SiStripDetId::ModuleGeometry moduleGeometry(const DetId &id) const; 
+  
+  // Represents a bitmask that is only valid in some subdetector. 
+  // However, checking validity still has to be done by the caller.
+  struct BitMask {
+    uint32_t startBit_;
+    uint32_t mask_;
+    int32_t  subdet;
+    uint32_t apply(DetId id) const { return ((id.rawId()>>startBit_) & mask_); };
+    bool     valid(DetId id) const { return id.subdetId() == subdet; };
+  };
+
+  std::map<std::string, BitMask> namedPartitions() const { 
+    std::map<std::string, BitMask> namedPartitions_{
```

Comment:
This should probably be pxNamedPartitions or pixelNamedPartitions.
Also, the method is rather costly and depends only on unchanging data members. So, it can be computed in the constructor.

It's unclear overall, if this implementation (esp the string names) is rather unique.
I think that it may be better to give access to pbVals_ and pfVals_  (``` PixelBarrelValues const & pbVals() const { return pbVals_; } ``` and let the DQM code come up with naming format at will.